client/packages/common/src/hooks/useDialog/useDialog.tsx

```diff
@@ -149,24 +160,53 @@ export const useDialog = (dialogProps?: DialogProps): DialogState => {
       animationTimeout
     );
 
+    const defaultPreventedOnClick =
+      (onClick: (e?: OkClickEvent) => Promise<boolean>) =>
+      (e?: OkClickEvent) => {
+        e && e.preventDefault();
+        return onClick(e);
+      };
+
     let WrappedNextButton: ModalProps['nextButton'] = undefined;
+    let WrappedOkButton: ModalProps['okButton'] = undefined;
+
     if (nextButton) {
-      const { onClick, ...restOfNextButtonProps } = nextButton.props;
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const { onClick, type, ...restOfNextButtonProps } = nextButton.props;
+
+      const handler = defaultPreventedOnClick(onClick);
 
       // TODO: If you want to change the slide direction or other animation details, add a prop
       // slideAnimationConfig and add a parameter to `useSlideAnimation` to pass in the config.
       WrappedNextButton = React.cloneElement(nextButton, {
         onClick: slideAnimation
-          ? async () => {
-              const result = await onClick();
+          ? async (e?: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+              const result = await handler(e);
               if (!!result) onTriggerSlide();
               return result;
             }
-          : onClick,
+          : handler,
+        type: !disableOkKeyBinding ? 'submit' : 'button',
         ...restOfNextButtonProps,
       });
     }
 
+    if (okButton) {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const { onClick, type, ...restOfOkButtonProps } = okButton.props;
+
+      WrappedOkButton = React.cloneElement(okButton, {
+        onClick: defaultPreventedOnClick(onClick),
+        // If the next button is not present/disabled, the ok button should be a submit button (allow firing on enter key press)
+        type:
+          !disableOkKeyBinding && (!nextButton || nextButton.props.disabled)
```

Comment:
curious what you think of enabling 'OK on enter' if the next button is disabled? The use case was going through existing stocktake/shipment lines, you could hit Enter to OK & Next your way through them, and then on the last item, when next wasn't available, Enter would fire the OK and close the modal

Reply:
I like it. and at the same time feel vaguely that I should be concerned that it isn't consistent or something.

As a user, the behaviour you have is perfect - when running through a list of things, you would be clicking OK & Next repeatedly until then end when you click OK. so yep, all good.

Reply:
> I like it. and at the same time feel vaguely that I should be concerned that it isn't consistent or something.

My sentiments exactly 😁